src/valid.rs

```diff
@@ -306,9 +299,9 @@ mod tests {
     #[test]
     fn test_trace() {
         let result = Valid::<(), i32, String>::fail(1)
-            .trace("A".into())
-            .trace("B".into())
-            .trace("C".into());
+            .trace("A".to_string())
+            .trace("B".to_string())
+            .trace("C".to_string());
```

Comment:
```suggestion
            .trace("C");
```
Should not be required I suppose.